### ISSUE TITLE
Need API documentation

### ISSUE THREAD

**Issue by jockm** [cesanta/mjs#98]
Aside from some examples in the readme, and the intro article, there is no documentation for the API.  It was only by examining the unit tests, `mjs_builtin.c`, and reading the v7 documentation through the internet archive, that we were able to figure out how to set and get properties in the VM and make calls into javascript functions with complex values.

This is less than ideal

**Comment by tripflex** [comment]
I agree, would be great to have some kind of available documentation to know what is, and is not available, even if it was something as simple as a list of available fn or commands that we could search through .... or maybe even later on convert into a custom eslint rules plugin for vscode or whatnot ... would be AMAZING to have and make it much easier for JS devs to dive into MOS and mJS 👍
